### DISCUSSION TITLE
Where to Change Default Context Length (num_ctx) from 2048 in Source Code

### DISCUSSION THREAD

**Discussion by Zayuki** [open-webui/open-webui#6402] [Q&A] [answered]
Hello everyone,

I’m in the process of setting up Open WebUI with Nginx as a reverse proxy for my company’s internal use. Since many of my colleagues aren’t experts in large language models (LLMs), I’d like to simplify their experience by increasing the default context length (num_ctx) from 2048 to 4096 tokens.

I believe that extending the default context length will improve usability by allowing for more detailed conversations without the need for frequent adjustments. However, I’m unsure where this default value is defined in the source code or how to effectively change it.

I’ve already attempted modifying the num_ctx value in open-webui-0.3.32/src/lib/components/chat/Settings/General.svelte and open-webui-0.3.32/src/lib/components/chat/Settings/Advanced/AdvancedParams.svelte, then rebuilding the container, but it didn’t work. The value of context length shown in Settings > General > Show Advanced Params.

Could someone guide me on where to find and modify this setting in the codebase?

**Comment by oatmealm** [comment]
If I'm not mistaken you can use a Modfile with the required defaults and use that model instead.

**Reply by Zayuki** [reply]
> If I'm not mistaken you can use a Modfile with the required defaults and use that model instead.

@oatmealm Hi, could you please clarify whether you're referring to a Modfile by Open WebUI, or did you mean a ModelFile from Ollama?

**Reply by tunganhem2018** [reply, marked as answer]
I think you create an custom model. Set parameters as you want and whitelist it for user use![image](https://github.com/user-attachments/assets/98604090-1c6d-4c7d-93da-84fe1a50a99d)

**Reply by Zayuki** [reply]
> I think you create an custom model. Set parameters as you want and whitelist it for user use![image](https://private-user-images.githubusercontent.com/43602365/380511465-98604090-1c6d-4c7d-93da-84fe1a50a99d.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwODQxMTAsIm5iZiI6MTczMDA4MzgxMCwicGF0aCI6Ii80MzYwMjM2NS8zODA1MTE0NjUtOTg2MDQwOTAtMWM2ZC00YzdkLTkzZGEtODRmZTFhNTBhOTlkLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyOFQwMjUwMTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Mjk3ZDIyMDkyY2ViY2ZkODY1NzIwZmRmNWMxN2QxZDJkMGU5Nzc3YTFhNGRmZTM1YTRkZjE3MTM2NzM4ZWMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WafhGjYpsQKbMpm4nJdm_sgnjHqHEB5UO4Oix8q__94)

Ohh ya. Thanks a lot.

**Reply by oatmealm** [reply]
> > If I'm not mistaken you can use a Modfile with the required defaults and use that model instead.
> 
> 
> 
> @oatmealm Hi, could you please clarify whether you're referring to a Modfile by Open WebUI, or did you mean a ModelFile from Ollama?

I typically create it within openwebui models section since there's a handy gui for that. ![image](https://github.com/user-attachments/assets/2bd1f773-d974-4060-ac67-471fe519405c)

**Comment by Daryl149** [comment]
This does not answer the initial question unfortunately: where to edit it in the source code? Since I delete and create the containers frequently for updates through docker, it's less work to edit it in the source code, so it sticks, instead of having to set it up in the UI after every deployment.

**Reply by rgaricano** [reply]
https://github.com/open-webui/open-webui/blob/b03fc97e287f31ad07bda896143959bc4413f7d2/src/lib/components/chat/Settings/Advanced/AdvancedParams.svelte#L970

(and rebuild)

**Reply by Daryl149** [reply]
That does not work unfortunately, like the original post said.
Changing this hard-coded value does not change the num_ctx being used by the models on rebuild. You then still need to click on the Controls section in the UI and click on Context Length to set it from Default to Custom to make the system use it.

Steps to reproduce:

- Change hard coded value at line 970 from 2048 to 100000
- Rebuild front end and back-end
- Upload large text document in UI with roughly 90000 tokens
- choose a model like llama3:8b
- click on the document and toggle it from 'Using Focused Retrieval' to 'Using Entire Document'
- ask question: Cite me the first words in this text.
- it take a very short time to answer. But it will fail to find the actual first words in the document, and instead cite some words near the end.
- click on Controls in the UI
- click on Context Length so it switches from Default to Custom
- close Controls
- press the Regenerate button
- it will now take a lot longer to answer and when it does, it will successfully find the first words in the document.

I think the real num_ctx at rebuild is saved somewhere in the database at creation time.
